greengrass_defender_agent/agent.py

```diff
@@ -125,7 +197,21 @@ def main():
     metrics_collector = collector.Collector(short_metrics_names=False)
 
     # Start collecting and publishing metrics
-    set_configuration_and_publish(ipc_client, configuration, metrics_collector)
+    publish_retry = env_config[config.CONNECT_AND_PUBLISH_RETRY_CONFIG_KEY]
```

Comment:
The publish to IoT core retry count will still be configurable to customer and as discussed with @youtuyy previously we're gonna load it from system env config.